agent/util-scripts/pbench-tool-meister

```diff
@@ -671,162 +668,180 @@ class ToolMeister(object):
-                                            response.text,
-                                        )
-                                        try:
-                                            shutil.rmtree(
-                                                self._tmp_dir / self._hostname
-                                            )
-                                        except Exception:
-                                            self.logger.exception(
-                                                "Failed to remove tool data hierarchy, '%s'",
-                                                self._tmp_dir / self._hostname,
-                                            )
-                                            failures += 1
-                            self.logger.debug(
-                                "%s: send_tools completed",
-                                self._hostname,
-                                self._group,
-                                self._directory,
-                            )
-                except Exception:
-                    self.logger.exception("Unexpected error encountered")
-            finally:
-                try:
-                    tar_file.unlink()
-                except OSError as exc:
-                    if exc.errno != errno.ENOENT:
-                        self.logger.warning(
-                            "error removing tools tar ball, '%s': %s", tar_file, exc
+                                        del self.directories[directory]
+                        self.logger.info(
+                            "%s: send_tools completed %s %s",
+                            self._hostname,
+                            self._group,
+                            tool_dir,
                         )
-                except Exception as exc:
+            except Exception:
+                self.logger.exception("Unexpected error encountered")
+        finally:
+            # We always remove the created tar file regardless of success or
+            # failure. The above code should take care of removing the
+            # directory tree the tar file was created from when it was
+            # successfully transferred, but if the transfer failed, we'll
+            # still have the local directory, so the tar file is still
+            # deletable.
+            try:
+                tar_file.unlink()
+            except OSError as exc:
+                if exc.errno != errno.ENOENT:
                     self.logger.warning(
-                        "unexpected error removing tools tar ball, '%s': %s",
-                        tar_file,
-                        exc,
+                        "error removing tools tar ball, '%s': %s", tar_file, exc
                     )
-        self._directory = None
+            except Exception as exc:
+                self.logger.warning(
+                    "unexpected error removing tools tar ball, '%s': %s", tar_file, exc,
+                )
```

Comment:
In terms of readability, even if the code is only used once, an argument could be made for refactoring the tar-ization and PUT into a separate method (or perhaps two separate methods) just because there's an urge to segment the code when reading; and finding the end takes effort. Not really important now, though.